lib/logidze/engine.rb

```diff
@@ -11,5 +12,11 @@ class Engine < Rails::Engine # :nodoc:
         ActiveRecord::Base.send :include, Logidze::HasLogidze
       end
     end
+
+    initializer "check Logidze function versions" do |app|
+      ActiveSupport.on_load(:active_record) do
+        app.config.app_middleware.use Logidze::Generators::CheckPending
```

Comment:
I think, it should only be included if pending check is enabled.

Also, doesn't it make sense to use some specific location in the middleware (e.g., `insert_after`).

See, for example:
https://github.com/rails/rails/blob/098fd7f9b3d5c6f540911bc0c17207d6b48d5bb3/activerecord/lib/active_record/railtie.rb#L86-L91

Reply:
Sure, I'll move a check here.

As for `insert_after`, it was my first idea to insert this middleware right after `ActiveRecord::Migration::CheckPending`, but it may be removed (with `config.active_record.migration_error = false`), then Rails will raise an error https://github.com/rails/rails/blob/291a3d2ef29a3842d1156ada7526f4ee60dd2b59/actionpack/lib/action_dispatch/middleware/stack.rb#L162

`app.config.app_middleware.use` pushes middleware to the end of the middleware list. I think it's good enough, or are there any pitfalls for this approach?

Reply:
> are there any pitfalls for this approach?

Can't come up with any. Let's leave `use`.